src/preferences/PreferencesBase.js

```diff
@@ -189,7 +198,13 @@ define(function (require, exports, module) {
                         try {
                             result.resolve(JSON.parse(text));
                         } catch (e) {
-                            result.reject(new ParsingError("Invalid JSON settings at " + path + "(" + e.toString() + ")"));
+                            if (createIfNew) {
+                                // JSON parsing error -- start from scratch
+                                console.log("Invalid JSON settings at " + path + "(" + e.toString() + ") -- contents reset");
+                                result.resolve({});
```

Comment:
This blows away all data. We could also try to move file to trash here in case user wants to try to retrieve some data.